### DISCUSSION TITLE
在没有bind接口的情况下，如何no-cache

### DISCUSSION THREAD

**Discussion by kulongwangzhi85** [pymumu/smartdns#1150] [Q&A]
只是在smartdns配置文件内部，使用domain-rules指定不同的nameserver，但想关闭该rules下的缓存。

**Comment by PikuZheng** [comment]
将ttl设得很小并关闭预读？

**Reply by kulongwangzhi85** [reply]
就是不想改全局的！

**Reply by kulongwangzhi85** [reply]
> 将ttl设得很小并关闭预读？

目前我只能再绑定个端口，并加-no-cache参数。但客户端请求还是走原来的53端口。不知有没效果

**Reply by kulongwangzhi85** [reply]
从日志看，好像不行，还是会从缓存读取。看来缓存是跟bind相关联的。

**Comment by PikuZheng** [comment]
>就是不想改全局的！

但是即使你在smartdns不缓存了，上游服务器也是会缓存的啊，ttl是由soa的设置决定的

**Reply by kulongwangzhi85** [reply]
需求是上游指向v2ray 使用了fakedns。

**Reply by PikuZheng** [reply]
fakettl本来就是1，在这种情况下关闭域名预解析（预读）缓存1秒就没了

**Reply by PikuZheng** [reply]
~猜测你的rr-ttl-min设的太大~应该设smartdns的rr-ttl-min=1

**Reply by kulongwangzhi85** [reply]
v2ray 的fake dns不知道是不是标准的，目前也没出现外网不能长时间访问的情况，只有重启smartdns后会出现一会儿无法访问。
再者，我们使用smartdns就是想预解析，来加快后续的访问。而且我们家用，访问国内的要比访问外网的要多得多。
所以关闭全局预解析，得不偿失。

**Reply by PikuZheng** [reply]
所以如果能按域名实现不缓存，是比较好的方案？

**Reply by kulongwangzhi85** [reply]
domain-rules

**Comment by pymumu** [comment]
没看懂，什么场景要关闭特定域名的缓存

**Reply by PikuZheng** [reply]
他前端是个dns抢答（或者说污染），特定域名返回一个特定网段的随机地址，在路由表中指派这个网段走代理出去（出去后对端会根据请求域名重新解析）。但是这个ip如果缓存了，就可能与抢答的不一致导致不通

**Reply by kulongwangzhi85** [reply]
不是抢答
**smartdns.conf**
```
server 127.0.0.1:15656 -group v2ray -exclude-default-group
domain-set -name gfwset -type list -file /etc/smartdns/gfw.txt
domain-rules /domain-set:gfwset/ -c none
nameserver /domain-set:gfwset/v2ray

domain-set -name greatfireset -type list -file /etc/smartdns/greatfire.txt
domain-rules /domain-set:greatfireset/ -c none
nameserver /domain-set:greatfireset/v2ray
```
![image](https://user-images.githubusercontent.com/13237693/200476868-a68718ea-f4db-4ac8-8dd5-20b15b95a664.png)

**Reply by PikuZheng** [reply]
那这个198.18.0.1是不变的还是变化的？从源码上看fakedns的所有应答都是ttl=1，理论上设rr-ttl-min=1，即使反复的查询也是没问题的

**Reply by kulongwangzhi85** [reply]
上游的v2ray开缓存，会变，但变的频率不会很高。

**Reply by pymumu** [reply]
意思是想减少一层缓存？

**Comment by kulongwangzhi85** [comment]
> 那这个198.18.0.1是不变的还是变化的？从源码上看fakedns的所有应答都是ttl=1，理论上设rr-ttl-min=1，即使反复的查询也是没问题的

大佬问下改小ttl值，会不会导致，smartdns也更频繁的发起更新？因为开了serve-expired？

**Reply by PikuZheng** [reply]
rr-ttl-min不会导致所有域名的ttl减小

**Comment by kulongwangzhi85** [comment]
曲线救国：
再开个smatdns示例，专门用于上游fake ip情况。将ttl值调小。
```
server-name gfw
bind [::]:13353 
bind-tcp [::]:13353
tcp-idle-time 60
cache-size 10240
cache-persist no
prefetch-domain yes
serve-expired yes
serve-expired-ttl 600
serve-expired-reply-ttl 6
speed-check-mode none
force-AAAA-SOA no
rr-ttl-min 1
rr-ttl-max 20
rr-ttl-reply-max 20
max-reply-ip-num 1
response-mode fastest-response
log-level debug
log-file /dev/shm/gfw.log
log-size 10m
log-num 2
server 127.0.0.1:15656
```
